domenetjenester/iay/src/main/java/no/nav/foreldrepenger/abakus/iay/tjeneste/dto/iay/MapInntektsmeldinger.java

```diff
@@ -64,6 +67,34 @@
         .thenComparing(im -> im.getArbeidsforholdRef() == null ? null : im.getArbeidsforholdRef().getAbakusReferanse(),
             Comparator.nullsLast(Comparator.naturalOrder()));
 
+
+    public static InntektsmeldingerDto mapUnikeInntektsmeldingerFraGrunnlag(List<InntektArbeidYtelseGrunnlag> grunnlag) {
+        List<InntektsmeldingDto> inntektsmeldinger = mapUnikeInntektsmeldinger(grunnlag);
+        InntektsmeldingerDto inntektsmeldingerDto = new InntektsmeldingerDto();
+        inntektsmeldingerDto.medInntektsmeldinger(inntektsmeldinger);
+        return inntektsmeldingerDto;
+    }
+
+    private static List<InntektsmeldingDto> mapUnikeInntektsmeldinger(List<InntektArbeidYtelseGrunnlag> grunnlag) {
```

Comment:
Bør ikke dra opp grunnlag, skriv heller en spørring som tar distinct inntektsmelding.
I tillegg, holder det å hente inntektsmeldinger fra aktive grunnlag (dropp scan av alle inaktive)?

Reply:
Det vi bruker resultatet til er å finne ut om det er er framsatt refusjonskrav innan fristen på 3 måneder. For ein gitt virksomhet og ein gitt dato må vi derfor finne ut om det har blitt søkt refusjon for denne datoen innan fristen blant alle tidligere inntektsmeldinger. Finnes det ein god måte å gjere dette på utan å dra opp alle inaktive?